File: Backend/JAVA/application/src/main/java/com/superhero/domain/behavior/repository/lookup/IBaseHeroLookup.java
package com.superhero.domain.behavior.repository.lookup;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import com.superhero.domain.model.BaseHero;

@Repository
public interface IBaseHeroLookup extends JpaRepository<BaseHero, Long>{

    @Query(value = "SELECT * FROM Base_Hero WHERE public_Id = ?1", nativeQuery = true)
    BaseHero getBaseHeroByPublicId(@Param("publicId") String publicId);

}

File: Backend/C#/SuperHero.Domain/Behavior/Repository/IBaseHeroLookup.cs
using SuperHero.Domain.Model;

namespace SuperHero.Domain.Behavior.Repository
{
    public interface IBaseHeroLookup
    {
        Task<(IEnumerable<BaseHero>, int TotalPages)> ListAllHeroesPaginated(int page, int pageSize, CancellationToken cancellationToken);

        Task<BaseHero> GetBaseHeroByPublicId(string publicId);
    }
}
